### PR TITLE
Initial pass at documentation

### PR DESCRIPTION
**Pull request by laconc**
Adding enough documentation to resolve https://github.com/datadotworld/dwapi-go/issues/4.

Also renaming `client` to `dw` in the unit tests to better match the user-facing experience.

### CODE REVIEW

File: dwapi/datasets.go [hunk 1]
```diff
@@ -59,58 +72,93 @@ func (s *DatasetService) CreateOrReplace(owner, id string, body *DatasetReplaceR
...
+//
+// Prefer `Query.ExecuteSQL()` or `Query.ExecuteSPARQL()` for retrieving clean and structured data.
 func (s *DatasetService) Download(owner, datasetid, filename string) (response io.Reader, err error) {
 	return s.client.File.DownloadDataset(owner, datasetid)
 }
 
+// DownloadAndSave downloads a .zip file containing all files within a dataset as originally
+// uploaded, and saves the results to a file.
+//
+// Prefer `Query.ExecuteSQL()` or `Query.ExecuteSPARQL()` for retrieving clean and structured data.
```
Comment (chrisbodhi) [inline dwapi/datasets.go]: Also known as "beautiful" data? 😀

File: dwapi/datasets.go [hunk 2]
```diff
@@ -59,58 +72,93 @@ func (s *DatasetService) CreateOrReplace(owner, id string, body *DatasetReplaceR
...
 }
 
+// Liked lists the datasets that the currently authenticated user has liked (bookmarked).
 func (s *DatasetService) Liked() (response []DatasetSummaryResponse, err error) {
 	return s.client.User.DatasetsLiked()
 }
 
+// ListQueries lists the saved queries associated with a dataset.
+//
+// Query definitions will be returned, not the query results. To retrieve the query results
```
Comment (chrisbodhi) [inline dwapi/datasets.go]: >query results **,** use

File: dwapi/datasets.go [hunk 3]
```diff
@@ -59,58 +72,93 @@ func (s *DatasetService) CreateOrReplace(owner, id string, body *DatasetReplaceR
...
 
+// Owned lists the datasets that the currently authenticated user has access to because they are
+// the owner.
 func (s *DatasetService) Owned() (response []DatasetSummaryResponse, err error) {
 	return s.client.User.DatasetsOwned()
 }
 
+// Retrieve fetches a dataset.
+//
+// The definition of will be returned, not the associated data. Use `Query.ExecuteSQL()`
```
Comment (chrisbodhi) [inline dwapi/datasets.go]: "The definition of" what "will be returned"?

File: dwapi/files.go [hunk 4]
```diff
@@ -84,13 +105,33 @@ func (s *FileService) DownloadAndSaveDataset(owner, id, path string) (response S
...
+		return
+	}
+
+	response, err = s.UploadStream(owner, id, filename, f, expandArchive)
+	if err != nil {
+		return
+	}
+
+	err = f.Close()
+	return
```
Comment (chrisbodhi) [inline dwapi/files.go]: Does this need to communicate to the user if something has failed, or is that handled by whatever calls this function?

File: dwapi/projects.go [hunk 5]
```diff
@@ -51,34 +62,51 @@ func (s *ProjectService) CreateOrReplace(owner, projectid string, body *ProjectC
...
 
+// DownloadAndSave downloads a .zip file containing all files within a project as originally
+// uploaded, and saves the results to a file.
+//
+// Prefer `Query.ExecuteSQL()` or `Query.ExecuteSPARQL()` for retrieving clean and structured data.
 func (s *ProjectService) DownloadAndSave(owner, projectid, path string) (response SuccessResponse, err error) {
 	return s.client.File.DownloadAndSaveDataset(owner, projectid, path)
 }
 
+// Liked lists the projects that the currently authenticated user has liked (bookmarked).
```
Comment (chrisbodhi) [inline dwapi/projects.go]: Product team should get this terminology consistent.

File: dwapi/streams.go [hunk 6]
```diff
@@ -41,20 +56,33 @@ func (s *StreamService) Append(owner, id, streamid string, body io.Reader) (resp
...
 
+// SetOrUpdateSchema sets or updates a stream’s schema. The schema of a stream defines its primary
+// key(s) and sort/sequence field.
+//
+// data.world streams are append-only by default. Alternatively, if a primary key is specified,
+// data.world will replace records with the same primary key value. data.world will sort records
+// by sequence field value and will discard all but the last record appended for each given
+// primary key value.
+//
+//The updateMethod parameter specifies how data.world should handle existing records when the
```
Comment (chrisbodhi) [inline dwapi/streams.go]: `//The` needs a space before `The`.

File: dwapi/webhooks.go [hunk 7]
```diff
@@ -33,27 +34,35 @@ func (s *WebhookService) List() (response []Subscription, err error) {
 	return
 }
 
+// RetrieveAccountSubscription fetches the webhook subscription associated with the currently
+// authenticated user and to the given organization or user account.
```
Comment (chrisbodhi) [inline dwapi/webhooks.go]: >webhook subscription associated with the currently authenticated user and to the given...

I think this sentence fragment could be more clear -- the mix of prepositions (`with` and `to`, specifically) tripped me up.

Maybe "webhook subscription based on the currently authenticated user and the given..."?

File: dwapi/datasets.go (hunk 2 above)
Comment (laconc) [reply dwapi/datasets.go]: Comma added!

File: dwapi/datasets.go (hunk 3 above)
Comment (laconc) [reply dwapi/datasets.go]: Lmao, fixed!

File: dwapi/projects.go (hunk 5 above)
Comment (laconc) [reply dwapi/projects.go]: Bookmarked is really what we're using now but unfortunately, `liked` is still everywhere in the API. I'll keep it as is and I imagine the language will be updated when we move to API version 1.

File: dwapi/files.go (hunk 4 above)
Comment (laconc) [reply dwapi/files.go]: The latter. If there's an error, it's passed back to the calling function and it has the responsibility of dealing with errors.

**Comment by laconc** [comment]
@chrisbodhi I merged master to get the tests running and fixed a few merge conflicts; would you mind taking a quick peek at some point?

**Pull request merged**